src/supply_info.rs

```diff
@@ -80,22 +79,30 @@ impl Default for SupplyInfo {
 mod tests {
     use super::*;
 
-    fn create_random_asset(seed: u64) -> AssetBase {
-        use {
-            group::{Group, GroupEncoding},
-            pasta_curves::pallas::Point,
-            rand::{rngs::StdRng, SeedableRng},
-        };
+    fn create_random_asset(asset_desc: &str) -> AssetBase {
+        use crate::keys::{IssuanceAuthorizingKey, IssuanceValidatingKey, SpendingKey};
 
-        AssetBase::from_bytes(&Point::random(StdRng::seed_from_u64(seed)).to_bytes()).unwrap()
+        let sk = SpendingKey::from_bytes([0u8; 32]).unwrap();
+        let isk: IssuanceAuthorizingKey = (&sk).into();
+
+        AssetBase::derive(&IssuanceValidatingKey::from(&isk), asset_desc)
+    }
+
+    fn sum_amounts<'a, Supplies: IntoIterator<Item = &'a AssetSupply>>(
+        supplies: Supplies,
+    ) -> Option<ValueSum> {
+        supplies
+            .into_iter()
+            .map(|supply| supply.amount)
+            .try_fold(ValueSum::from_raw(0), |sum, value| sum + value)
```

Comment:
Very nice, 
I would remove repeated terms to allow the code speak for itself:
```rust
    fn sum<'a, T: IntoIterator<Item = &'a AssetSupply>>(
        supplies: T,
    ) -> Option<ValueSum> {
        supplies
            .into_iter()
            .map(|supply| supply.amount)
            .try_fold(ValueSum::from_raw(0), |sum, value| sum + value)
    }
```
A `sum` over `supplies`.